data/data/bootstrap/files/usr/local/bin/installer-gather.sh

```diff
@@ -9,54 +9,108 @@ fi
+    mkdir -p "${ARTIFACTS}/unit-status"
+    sed -n 's/^\* \([^ ]*\) .*/\1/p' < "${ARTIFACTS}/failed-units.txt" | while read -r UNIT
+    do
+        systemctl status --full "${UNIT}" >& "${ARTIFACTS}/unit-status/${UNIT}.txt"
+        journalctl -u "${UNIT}" > "${ARTIFACTS}/unit-status/${UNIT}.log"
+    done
+
+    echo "Gathering bootstrap containers ..."
+    mkdir -p "${ARTIFACTS}/bootstrap/containers"
+    sudo crictl ps --all --quiet | while read -r container
+    do
+        container_name="$(sudo crictl ps -a --id "${container}" -v | grep -oP "Name: \\K(.*)")"
+        sudo crictl logs "${container}" >& "${ARTIFACTS}/bootstrap/containers/${container_name}-${container}.log"
+        sudo crictl inspect "${container}" >& "${ARTIFACTS}/bootstrap/containers/${container_name}-${container}.inspect"
+    done
+    sudo cp -r /var/log/bootstrap-control-plane/ "${ARTIFACTS}/bootstrap/containers"
+    mkdir -p "${ARTIFACTS}/bootstrap/pods"
+    sudo podman ps --all --quiet | while read -r container
+    do
+        sudo podman logs "${container}" >& "${ARTIFACTS}/bootstrap/pods/${container}.log"
+        sudo podman inspect "${container}" >& "${ARTIFACTS}/bootstrap/pods/${container}.inspect"
+    done
+
+    echo "Gathering rendered assets..."
+    mkdir -p "${ARTIFACTS}/rendered-assets"
+    sudo cp -r /var/opt/openshift/ "${ARTIFACTS}/rendered-assets"
+    sudo chown -R "${USER}":"${USER}" "${ARTIFACTS}/rendered-assets"
+    sudo find "${ARTIFACTS}/rendered-assets" -type d -print0 | xargs -0 sudo chmod u+x
+    # remove sensitive information
+    # TODO leave tls.crt inside of secret yaml files
+    find "${ARTIFACTS}/rendered-assets" -name "*secret*" -print0 | xargs -0 rm -rf
+    find "${ARTIFACTS}/rendered-assets" -name "*kubeconfig*" -print0 | xargs -0 rm
+    find "${ARTIFACTS}/rendered-assets" -name "*.key" -print0 | xargs -0 rm
+    find "${ARTIFACTS}/rendered-assets" -name ".kube" -print0 | xargs -0 rm -rf
+
+    echo "Gather remote logs"
+    export MASTERS=()
+    if [ "$#" -ne 0 ]; then
+        MASTERS=( "$@" )
+    elif test -s "${ARTIFACTS}/resources/masters.list"; then
+        mapfile -t MASTERS < "${ARTIFACTS}/resources/masters.list"
+    else
+        echo "No masters found!"
+    fi
+
+    for master in "${MASTERS[@]}"
+    do
+        echo "Collecting info from ${master}"
+        scp -o PreferredAuthentications=publickey -o StrictHostKeyChecking=false -o UserKnownHostsFile=/dev/null -q /usr/local/bin/installer-masters-gather.sh "core@[${master}]:"
+        mkdir -p "${ARTIFACTS}/control-plane/${master}"
+        ssh -o PreferredAuthentications=publickey -o StrictHostKeyChecking=false -o UserKnownHostsFile=/dev/null "core@${master}" -C "sudo ./installer-masters-gather.sh --id '${GATHER_ID}'" </dev/null
+        scp -o PreferredAuthentications=publickey -o StrictHostKeyChecking=false -o UserKnownHostsFile=/dev/null -r -q "core@[${master}]:/tmp/artifacts-${GATHER_ID}/*" "${ARTIFACTS}/control-plane/${master}/"
+    done
```

Comment:
This needs to stay at the bottom as `resources/masters.list` is populated by `Gathering cluster resources ...`. I'll leave a comment about reworking this logic below.